doc/devel/development_setup.rst

```diff
@@ -4,6 +4,51 @@
 Setting up Matplotlib for development
 =====================================
 
+.. note::
+
+    Setting up everything correctly and working with git can be challenging for
+    new contributors. If you have difficulties, feel free to ask for help. We
+    have a dedicated :ref:`contributor-incubator` channel on gitter for
+    onboarding new contributors.
+
+Setting up git and retrieving the latest version of the code
+============================================================
+
+Matplotlib is hosted on `GitHub <https://github.com/matplotlib/matplotlib>`_.
+To contribute you will need to sign up for a `free GitHub account
+<https://github.com/signup/free>`_.
+
+GitHub uses Git for version control, which allows many people to work together
+on the project. See the following links for more information:
+
+- `GitHub help pages <https://help.github.com/>`_
+- `NumPy documentation <https://numpy.org/doc/stable/dev/index.html>`_
+- `pandas documentation <https://pandas.pydata.org/docs/development/contributing.html#working-with-the-code>`_
+
+Forking
+-------
+Go to https://github.com/matplotlib/matplotlib and click the `Fork` button in
+the top right corner to create your own copy of the project.
+
+Now clone your fork to your machine:
+
+.. code-block:: bash
+
+    git clone https://github.com/your-user-name/matplotlib.git
+    cd matplotlib
+    git remote add upstream https://github.com/matplotlib/matplotlib.git
+
+This creates a directory `matplotlib` under your current working directory
```

Comment:
Also would break this out. Creates a directory: clone command

then we go into the directory: cd command (link out to command line resource) <- also cd doesn't work on windows, is dir

then we connect: git add remote